spec/features/admin_creates_genres_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+describe 'As an admin user' do
+  context 'when I visit the genre index' do
+    it 'I can fill out a form to create a new genre' do
+      admin = User.create!(username: "user", password: "pass", role: "admin")
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+
+      visit(genres_path)
+
+      fill_in 'Name', with: 'Action'
+      click_on 'Submit'
+
+      expect(current_path).to eq(genres_path)
+      expect(page).to have_content('Action')
+    end
```

Comment:
ruh-roh, no test for the sad-path of a genre not saving ... which would have caught the problem I mentioned earlier in your GenreController with the missing instance variable.